deploy/job-manager/deploy.sh

```diff
@@ -0,0 +1,329 @@
+#!/usr/bin/env bash
+
+# This script is designed to be run by Jenkins to deploy a new Job Manager kubernetes deployment
+# =======================================
+# Example Usage:
+# bash deploy.sh dev v0.0.4 username password
+# =======================================
+
+function line() {
+    echo -ne "=======================================\n"
+}
+
+# Tested
+function stdout() {
+    local msg=$1
+    echo -ne ">| Current Step |< ${msg} ...\n"
+}
+
+# Tested
+function stderr() {
+    echo -ne ">>| Error Occurred |<< - exiting ...\n"
+    exit 1
+}
+
+# Tested
+function configure_mint_kubernetes() {
+    local ENV=$1
+
+    stdout "Setting to use Google project: project broad-dsde-mint-${ENV}"
```

Comment:
This can be generalized by passing in the gcloud project and gke context instead of hard-coding it to `broad-dsde-mint`

Reply:
Yeah, that would be much better. The issue here is this deployment script not only uses GKE, but also GCR, we need to make sure the target project satisfies all the requirements. So I scoped down this PR to only focus on polishing the Job Manager deploy script for Mint's internal uses. I plan to leave that part of work to the future when we want to merge this script with the "Green Box set up" script.